source/Octopus.Tentacle/Scripts/IScriptExecutor.cs

```diff
@@ -5,6 +5,7 @@ namespace Octopus.Tentacle.Scripts
 {
     public interface IScriptExecutor
     {
-        IRunningScript ExecuteOnBackgroundThread(StartScriptCommandV3Alpha command, IScriptWorkspace workspace, ScriptStateStore? scriptStateStore, CancellationToken cancellationToken);
+        bool ValidateExecutionContext(IScriptExecutionContext executionContext);
```

Comment:
Naming suggestion, we use a semantic like `CanExecute` in other places. It makes it a little clearer what the check is doing than just validate, which sounds like it could be verifying the contents itself rather than just checking if its valid to be passed in at all.

Reply:
Changed in [a2a48b7](https://github.com/OctopusDeploy/OctopusTentacle/pull/690/commits/a2a48b76b42da74a82b8d99115f1ce3d09c49010)

Now takes the entire command, rather than just the execution context